python/pyspark/ml/feature.py

```diff
@@ -775,7 +775,7 @@ def mean(self):
 
 
 @inherit_doc
-class StringIndexer(JavaEstimator, HasInputCol, HasOutputCol):
+class StringIndexer(JavaEstimator, HasInputCol, HasOutputCol, HasHandleInvalid):
```

Comment:
```handleInvalid``` is common properties and other ```Transformer/Estimator```s may use it in future, so we need to make it shared param. Another reason is that we want to make Python consistency with Scala, and [```handleInvalid```](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/param/shared/sharedParams.scala#L255) in Scala is shared param.
@jkbradley